lib/spack/spack/mixins.py

```diff
@@ -0,0 +1,82 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+
+import llnl.util.filesystem
+
+
+class FilterCompilerWrappersPackageMixin(object):
+    """This mixin class registers a callback that filters a list of files
+    after installation and substitutes hardcoded paths pointing to the Spack
+    compiler wrappers with the corresponding 'real' compilers.
+    """
+
+    #: compiler wrappers to be filtered (needs to be overridden)
+    compiler_wrappers = []
```

Comment:
Yeah, naming is kind of provisional... I am also not really satisfied by the name of the class. I think it would be fine if not exposed to packagers, but to somebody not used to programming `FilterCompilerWrappersPackageMixin` may be intimidating. I am definitely open for better names.